### PR TITLE
Migrate to core kit

### PR DESCRIPTION
**Pull request by arnaldo2792**
**Issue number:**

Related to #4048

**Description of changes:**

This PR migrates the code to use the Bottlerocket core kit.

**Testing done:**
- Built an AMI with my personal core kit, and verified the host booted.

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Review by cbgbt — COMMENTED** [review]
Nice! :rocket: 

This looks good to me so far, I'll take another look when the tests are passing.

File: sources/Cargo.toml [hunk 1]
```diff
@@ -1,26 +1,9 @@
...
-    "api/sundog",
-    "api/schnauzer",
-    "api/pluto",
-    "api/host-containers",
-    "api/storewolf",
-    "api/thar-be-settings",
-    "api/thar-be-updates",
-    "api/settings-committer",
-    "api/migration/migrator",
+    "api/apiclient",
```
Comment (sam-berning) [inline sources/Cargo.toml:5]: nit: this is out of alphabetical order with datastore

File: sources/Cargo.toml [hunk 2]
```diff
@@ -113,22 +69,7 @@ members = [
...
-    "parse-datetime",
-
-    "retry-read",
-
-    "updater/block-party",
-    "updater/signpost",
-    "updater/update_metadata",
-    "updater/updog",
-
     "constants",
```
Comment (sam-berning) [inline sources/Cargo.toml:72]: duplicate?

File: variants/aws-ecs-1-nvidia/Cargo.toml [hunk 3]
```diff
@@ -39,3 +39,4 @@ path = "../variants.rs"
 [build-dependencies]
 settings-defaults = { path = "../../packages/settings-defaults" }
 settings-plugins = { path = "../../packages/settings-plugins" }
+settings-migrations = { path = "../../packages/settings-migrations" }
```
Comment (sam-berning) [inline variants/aws-ecs-1-nvidia/Cargo.toml:42]: nit: out of alphabetical order

**Review by sam-berning — APPROVED** [review]

**Comment by bcressey** [comment]
Looks good to me once we have the first core kit release.

**Review by webern — APPROVED** [review]

**Review by cbgbt — APPROVED** [review]

**Review by bcressey — APPROVED** [review]

**Review by cbgbt — APPROVED** [review]
:rocket:

**Review by sam-berning — APPROVED** [review]

**Pull request merged**